js/filters.js

```diff
@@ -115,9 +145,27 @@ function TypeaheadFilter(elm) {
   var key = this.$body.data('dataset');
   var dataset = typeahead.datasets[key];
   this.typeaheadFilter = new typeaheadFilter.TypeaheadFilter(this.$body, dataset);
+  this.typeaheadFilter.$body.on('change', 'input[type="checkbox"]', this.handleNestedChange.bind(this));
 }
 
 TypeaheadFilter.prototype = Object.create(Filter.prototype);
 TypeaheadFilter.constructor = TypeaheadFilter;
 
+TypeaheadFilter.prototype.handleChange = function() {};
+
+TypeaheadFilter.prototype.handleNestedChange = function(e) {
```

Comment:
This would go away if we decide to merge the two typeahead classes later on. But for now, I wanted to have `filter:added` and `filter:removed` events only emitted by `Filter` and its subclasses.